_source/_posts/2020-01-16-transactional-mfa.md

```diff
@@ -0,0 +1,265 @@
+
+Classic UI Console:
+
+{% img blog/transactional-mfa/classic-console.png alt:"Classic Console" width:"800" %}{: .center-image }
+
+Click **Applications** -> **Add Application**. Next, click **Add Application** -> **Create New App** -> **Select Web and SAML**.
+
+{% img blog/transactional-mfa/create-new-application.png alt:"Create New Application" width:"800" %}{: .center-image }
+
+Enter an **App name** value (I'm using `MFA as a Service`), then click **Next**.
+
+{% img blog/transactional-mfa/create-saml-integration-one.png alt:"Create SAML Integration Step One" width:"800" %}{: .center-image }
+
+Enter placeholder (`http://localhost:8080`) data on the **Single sign on URL**, **Audience URI** here as you actually will not use this data during the flow. This is required so that you can create the custom SAML application. Once done, click **Next**.
+
+{% img blog/transactional-mfa/create-saml-integration-two.png alt:"Create SAML Integration Step Two" width:"800" %}{: .center-image }
+
+In the last screen, you can select either radio button and click **Finish**.
+
+You will be redirected to the **Application Sign On Policy Tab**. Scroll down and look for **App Sign-On Policy** and click **Add Rule**. This will allow you to create a sign on rule to challenge MFA every time. Enter a name, and under **Actions**, select **Prompt for Factor** and select **Every Sign On**. Click **Save**.
+
+{% img blog/transactional-mfa/app-sign-on-rule.png alt:"App Sign On Rule" width:"800" %}{: .center-image }
+
+Click the **General** Tab and scroll down until you see the **App Embed Link** section and click **Edit**. Enter the URL where you will host a separate widget that will do MFA as a service. In this example, I've created a basic Node.JS web application and hosted it under `http://localhost:<port>`. Click **Save**. I will be using port 3000 for this blog post.
+
+{% img blog/transactional-mfa/app-embed-link.png alt:"App Embed Link" width:"800" %}{: .center-image }
+
+Take note of the embed link URL above which will be used later to trigger the transactional MFA process.
+
+## Create a NodeJS Express Application in Okta
+
+For the remaining steps, you will need to switch back to the Dev UI Console, then click **Applications** -> **Add Application**. Next, click **Single Page Application** -> **Next**.
+
+This will create an OpenID Connect application representing your Node.js application within Okta.
+
+{% img blog/transactional-mfa/create-node-app-okta.png alt:"Create Node Application" width:"800" %}{: .center-image }
+
+You'll be presented with a screen to configure your Okta OpenID Connect application where you can change the name to "NodeJS Express Application". As for the **Login redirect URIs** field, this will be the URL where you want to redirect the user to your authenticated page. (e.g. `http://localhost:<port>/implict/callback`). I will be using port 8080 here.
```

Comment:
```suggestion
You'll be presented with a screen to configure your Okta OpenID Connect application where you can change the name to "NodeJS Express Application". As for the **Login redirect URIs** field, this will be the URL where you want to redirect the user to your authenticated page. (e.g. `http://localhost:<port>/implicit/callback`). I will be using port 8080 here.
```